spec/rails_helper.rb

```diff
@@ -510,3 +511,32 @@ class << self
     attr_accessor :value
   end
 end
+
+class TestSecondFactorAction < SecondFactor::Actions::Base
+  def self.called_methods
```

Comment:
This is a bit of an odd pattern to me. It seems like you are re-implementing stubs yourself. Rather than doing this, could we not use `expects` and `stubs` in your tests?